apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -85,5 +92,114 @@ public void CloneUpdateWithParameters()
             Assert.True(command.Parameters.SequenceEqual(command2.Parameters));
         }
 
+        [Fact]
+        public void CommandHasConnectionQueryOptions()
+        {
+            const string optimizerVersion = "1";
+            Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(optimizerVersion);
+
+            SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+            var queryOptions = new QueryOptions().WithOptimizerVersion(optimizerVersion);
+            connection.QueryOptions = queryOptions;
+
+            var command = connection.CreateSelectCommand("SELECT * FROM FOO");
+            using (var reader = command.ExecuteReader())
+            {
+                // Do nothing.
+            }
+        }
+
+        [Fact]
+        public void CommandHasOptimizerVersionFromEnvironment()
+        {
+            // Save existing value of environment variable.
+            const string optimizerVersionVariable = "SPANNER_OPTIMIZER_VERSION";
+            string savedOptimizerVersion = Environment.GetEnvironmentVariable(optimizerVersionVariable);
+            const string envOptimizerVersion = "2";
+            Environment.SetEnvironmentVariable(optimizerVersionVariable, envOptimizerVersion);
+
+            // Optimizer version set through environment variable has higher
+            // precedence than version set through connection.
+            Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(envOptimizerVersion);
+
+            const string optimizerVersion = "1";
+            SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+            var queryOptions = new QueryOptions().WithOptimizerVersion(optimizerVersion);
+            connection.QueryOptions = queryOptions;
+
+            var command = connection.CreateSelectCommand("SELECT * FROM FOO");
+            using (var reader = command.ExecuteReader())
+            {
+                // Do nothing.
+            }
+
+            // Set the environment back.
+            Environment.SetEnvironmentVariable(optimizerVersionVariable, savedOptimizerVersion);
```

Comment:
It would be good to do this in a `finally` block. You might even want to write a method accepting an `Action` that will set the environment variable, execute the action, then reset the environment variable.